bayesian/api/api_v2.py

```diff
@@ -202,6 +205,55 @@ def get(ecosystem, package, version):
 
         raise HTTPError(404, msg)
 
+    @staticmethod
+    def post():
+        """Handle the POST REST API call.
+
+        Component Analyses Batch is a 3 Step Process:
+        1. Gather and clean Request.
+        2. Query GraphDB, Builds Stacks Recommendation, Get Unknown Packages Set
+            and Trigger componentApiFlow.
+        3. Handle Unknown Packages and Trigger bayesianApiFlow.
+        """
+        response_template: Tuple = namedtuple("response_template", ["message", "status"])
+        ingestion_disabled_msg: str = "No data found for any package in manifest file. " \
+                                      "Ingestion flow skipped as DISABLE_UNKNOWN_PACKAGE_FLOW " \
+                                      "is enabled"
+        no_package_available_msg: str = "No Package in given manifest is available. " \
+                                        "Packages will be available shortly, " \
+                                        "Please retry after some time."
+
+        # Step1: Gather and clean Request
+        input_json: Dict = request.get_json()
+        ecosystem: str = input_json.get('ecosystem')
+        try:
+            packages_list: List = ca_validate_input(input_json, ecosystem)
+        except BadRequest as br:
+            logger.error(br)
+            raise HTTPError(400, str(br)) from br
+
+        # Step2: Query GraphDB, Build Unknown packages set and Generates Stack Recommendation.
+        try:
+            stack_recommendation, unknown_pkgs = get_ca_batch_response(ecosystem, packages_list)
+        except Exception as e:
+            msg = "Internal Server Exception. Please contact us if problem persists."
+            logger.error(e)
+            raise HTTPError(400, msg)
+
+        # Step3: Handle Unknown Packages
+        if unknown_pkgs:
+            if os.environ.get("DISABLE_UNKNOWN_PACKAGE_FLOW") == "1":
+                # Unknown Packages is Present and INGESTION is DISABLED
+                logger.error(ingestion_disabled_msg)
+                raise HTTPError(400, ingestion_disabled_msg)
+            unknown_package_flow(ecosystem, unknown_pkgs)
+            if not stack_recommendation:
+                # If No Package is Known, and all are unknown.
+                logger.debug(no_package_available_msg)
+                return response_template({'error': no_package_available_msg}, 202)
```

Comment:
Any reason for using both response_template and HttpError for error?

Reply:
Actually, in our Monitoring Server, which may come in later, HttpError is taken as Exception, though it's really not an Exception, so I kept it as `response_template`

Changed it to HTTPError for now.